Allow client-side commands to run chat components

fi-xz opened the pull request:
Backporting #554 from 1.21.x to 1.21.1. While the patch is usable on 1.21.x branch it does not on 1.21.1 branch

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/neoforged/NeoForge?pullRequest=2323) <br/>All committers have signed the CLA.

Matyrobbrt commented on patches/net/minecraft/client/multiplayer/ClientPacketListener.java.patch:
```diff
@@ -138,6 +138,14 @@
+     }
+ 
+     public boolean sendUnsignedCommand(String p_251509_) {
++        if (net.neoforged.neoforge.client.ClientCommandHandler.runCommand(p_251509_)) return true;
```
Please keep the comment from the PR

fi-xz replied:
Missed that, sorry. Should be resolved with a new force-pushed commit soon.

fi-xz replied:
[`838a53e`](https://github.com/fi-xz/NeoForge/commit/838a53ef0595a1d358b72b2434ec481dba64f17a) fixes the problem. I will mark it as resolved, but in case if it's wrong feel free to point that out. 🙏

fi-xz committed:
Allow client-side command to be able to run on chat components

Backporting #554 from 1.21.x to 1.21.1. While the patch is usable on 1.21.x branch it does not on 1.21.1 branch

fi-xz commented:
Seems like I'm terrible at doing PR. Sorry for the GitHub Actions ran many times (and mentioning PR from commits) while the code itself is not following the upstream.

fi-xz commented:
Resolved an issue where original PR(554) comment was missing at this PR.

fi-xz commented:
Thank you! ❤️
